### PR TITLE
Bring CCDB more in line with O2 coding guidelines

### PR DESCRIPTION
**Pull request by Barthelemy**
To comply with coding guidelines.

### CODE REVIEW

File: CCDB/include/CCDB/FileStorage.h [hunk 1]
```diff
@@ -20,12 +20,12 @@ class TFile;  // lines 8-8
 class TList;
 
 class TObject;
-namespace o2 { namespace CDB { class Condition; }}
-namespace o2 { namespace CDB { class ConditionId; }}
-namespace o2 { namespace CDB { class IdRunRange; }}
+namespace o2 { namespace ccdb { class Condition; }}
+namespace o2 { namespace ccdb { class ConditionId; }}
+namespace o2 { namespace ccdb { class IdRunRange; }}
```
Comment (matthiasrichter) [inline CCDB/include/CCDB/FileStorage.h]: that could be one namespace block for all forward declarations

**Comment by matthiasrichter** [comment]
There are two macros failing now, I think the namespace needs to be adjusted there as well
```
  1 - /CCDB/example/fill_local_ocdb.C (Failed)
63 - /macro/putCondition.C (Failed)
```

Can you also also grep for `CDB::` to check if there are more cases like that not caught by compilation nor unit test.

**Comment by Barthelemy** [comment]
CCDB 
* fixed namespace (CDB->ccdb)
* fixed indentation in all files
* fixed precompiler instruction in Linkdef
* updated copyright headers
* fixed formatting in all files using clang-format

General
* fixed formatting section in README

**Comment by Barthelemy** [comment]
I don't know why the travis test fails. 
The failed test in macos seems unrelated.

**Comment by sawenzel** [comment]
@Barthelemy : The travis test fails because clang-format fails. Note that clang-format is only checked on files which have been touched. Basically you touched ccdb here which did not yet have good formatting.
Could you just correctly format the files? ... then it should be ok.

**Comment by Barthelemy** [comment]
@sawenzel Sorry for the delay, I was off for a week. 

Quick question concerning the formatting : 
 It seems that the pointer or reference character (* &) are expected to have no space with the type. Is that correct ? the coding guidelines do not force either syntax : https://rawgit.com/AliceO2Group/CodingGuidelines/master/naming_formatting.html?showone=Pointer_and_Reference_Expressions#Pointer_and_Reference_Expressions
I don't have a strong opinion but would like to clarify to adapt my IDE settings. 

I am fixing the other clang-format issues.

**Comment by Barthelemy** [comment]
@sawenzel done

**Comment by sawenzel** [comment]
I think having the place of &* consistent would be beneficial to read the code. And to be pragmatic I would suggest to take clang-format as reference, put the rule in the guidelines and use it also in IDEs.

**Comment by Barthelemy** [comment]
@sawenzel I agree. I will prepare a PR to modify the coding guidelines accordingly and let people debate on it.

**Pull request merged**